### PR TITLE
feat: add optional signing (defaults to false) of container

### PR DESCRIPTION
**Pull request by XSlemX**
- Signs container and verifies.  
- Added as opt in initially.

### CODE REVIEW

File: .github/workflows/reusable_publish.yaml [hunk 1]
```diff
@@ -134,7 +139,24 @@ jobs:
...
+      - uses: sigstore/cosign-installer@536b37ec5d5b543420bdfd9b744c5965bd4d8730 # tag=v2.3.0
+      - name: Sign the images
+        if: ${inputs.enable-container-signing}
+        run: |
+          cosign sign
+            -a "repo=${{ github.repository }}" \
+            -a "workflow=${{ github.workflow }}" \
+            -a "ref=${{ github.sha }}" \
+            ${TAGS}
+        env:
```
Comment (AuStien) [inline .github/workflows/reusable_publish.yaml:151]: Would it be possible to move the `env` field above `run`?
Confusing to read envvars in the `run`-block not yet defined :sweat_smile:

File: .github/workflows/reusable_publish.yaml [hunk 2]
```diff
@@ -134,7 +139,24 @@ jobs:
...
+          TAGS: ${{ steps.digest-prep.outputs.image }}
+          COSIGN_EXPERIMENTAL: 1
+      - name: Verify the pushed tags
+        if: ${inputs.enable-container-signing}
+        run: cosign verify ${TAGS}
+        env:
```
Comment (AuStien) [inline .github/workflows/reusable_publish.yaml:157]: See above

**Review by AuStien — COMMENTED** [review]
Very cool, one minor feedback

File: .github/workflows/reusable_publish.yaml (hunk 1 above)
Comment (XSlemX) [reply .github/workflows/reusable_publish.yaml:151]: Agreed 🙌

**Pull request merged**